### PR TITLE
[IMP] spreadsheet: update o_spreadsheet to latest version

### PR DESCRIPTION
**Pull request by rrahir**
### Contains the following commits:

https://github.com/odoo/o-spreadsheet/commit/9ceed961b [REL] 17.4.0-alpha.5 Task: 0
https://github.com/odoo/o-spreadsheet/commit/80adcebb9 [FIX] charts: fix scorecard chard design Task: 3964015
https://github.com/odoo/o-spreadsheet/commit/658c00be3 [REF] pivot: do not create table in INSERT_PIVOT command Task: 3987435
https://github.com/odoo/o-spreadsheet/commit/584ccbcb0 [FIX] Figure: Fix container size in presence of frozen panes Task: 3976086
https://github.com/odoo/o-spreadsheet/commit/645a800cd [IMP] locale: add warning Task: 3989485
https://github.com/odoo/o-spreadsheet/commit/d70c2d8a9 [REM] chart: remove unused code Task: 0
https://github.com/odoo/o-spreadsheet/commit/76448b329 [FIX] collaborative: snapshot only when no pending changes Task: 0

Co-authored-by: Anthony Hendrickx (anhe) <anhe@odoo.com>
Co-authored-by: Alexis Lacroix (laa) <laa@odoo.com>
Co-authored-by: Lucas Lefèvre (lul) <lul@odoo.com>
Co-authored-by: Dhrutik Patel (dhrp) <dhrp@odoo.com>
Co-authored-by: Adrien Minne (adrm) <adrm@odoo.com>
Co-authored-by: Mehdi Rachico (mera) <mera@odoo.com>
Co-authored-by: Rémi Rahir (rar) <rar@odoo.com>
Co-authored-by: Pierre Rousseau (pro) <pro@odoo.com>
Co-authored-by: Vincent Schippefilt (vsc) <vsc@odoo.com>

### CODE REVIEW

**Comment by rrahir** [comment]
robodoo fw=no

**Comment by pro-odoo** [comment]
robodoo rebase-merge

File: addons/spreadsheet/static/src/pivot/plugins/pivot_odoo_ui_plugin.js [hunk 1]
```diff
@@ -29,4 +36,33 @@ export class PivotOdooUIPlugin extends OdooUIPlugin {
...
+            right: position.col + pivotCells.length - 1,
+        };
+        const numberOfHeaders = table.columns.length - 1;
+        const cmdContent = {
+            sheetId: position.sheetId,
+            ranges: [this.getters.getRangeDataFromZone(position.sheetId, pivotZone)],
+            config: { ...PIVOT_TABLE_CONFIG, numberOfHeaders },
+            tableType: "static",
+        };
+        if (this.canDispatch("CREATE_TABLE", cmdContent).isSuccessful) {
```
[the remark this answers is not in the document]
Comment (pro-odoo) [reply addons/spreadsheet/static/src/pivot/plugins/pivot_odoo_ui_plugin.js]: I removed it

**Comment by rrahir** [comment]
@robodoo r+

**Pull request closed without merging**